### PR TITLE
Upgrade ESLint to v5

### PR DESCRIPTION
**Pull request by jeremywiebe**
Upgrade the main ESLint to v5. Upgrade all eslint pugins to latest
(including `babel-eslint`).

Fixed up eslint rules that are now deprecated or irrelevant due to our
usage of Prettier (`flowtype/boolean-style`, `no-unexpected-multiline`).

Sets `ecmaVersion` to 2018 as the experimental-object-rest-spread rule
is deprecated and support for object rest spread is introduced in
`ecmaVersion=2018`.

`ecmaFeatures` has been deprecated so that's also removed.

## Test Plan:
I connected this to our mobile repo on a branch prepping for an upgrade
to React Native 0.57 (includes a Babel 7, which is why this revision is
needed) and ran `arc lint --everything` without errors.

- [ ] Run `ka-lint` in webapp

### CODE REVIEW

**Comment by jeremywiebe** [comment]
Unfortunately neither of those solutions will work. We use `ka-lint` via `arc` in our mobile repo. The side effect of that is that `ka-lint` uses the ESLint that lives in the `khan-linter` directory instead of whatever's in the project. That's what triggered this entire pull request. 

I'll think about this more to see if there's a way to not break `webapp` but allow us to move forward.

**Comment by csilvers** [comment]
John and I have a philosophical disagreement here -- I don't want to see mobile and webapp using different linting rules.  The third solution is to get webapp to pass with the new version of eslint.  For all we know, it already does, or could be made to do so without too much effort.

He's right though that we shouldn't land this until we've verified that webapp, mobile, and any other repos we have using khan-linter (khanalytics?) pass after this upgrade.

**Comment by csilvers** [comment]
(To the extent that wonder-blocks is a third-party project, I am fine with it using different lint rules than Khan code.)

File: eslintrc [hunk 1]
```diff
@@ -29,19 +29,15 @@
         // TODO(csilvers): enable these if/when community agrees on it.
         "prefer-arrow-callback": 0
     },
-    "ecmaFeatures": {
```
Comment (kevinbarabash) [inline eslintrc]: Thanks for cleaning this up.

**Comment by jeremywiebe** [comment]
https://phabricator.khanacademy.org/D50090 landed and makes this PR irrelevant. :)

**Pull request closed without merging**